plugins/scaffolder-backend/src/scaffolder/actions/builtin/catalog/write.ts

```diff
@@ -19,32 +19,30 @@ import { createTemplateAction } from '../../createTemplateAction';
 import * as yaml from 'yaml';
 import { Entity } from '@backstage/catalog-model';
 import { resolveSafeChildPath } from '@backstage/backend-common';
+import { z } from 'zod';
+import { convertZodtoJson } from '../../../tasks/convertZodtoJson';
 
 /**
  * Writes a catalog descriptor file containing the provided entity to a path in the workspace.
  * @public
  */
+
 export function createCatalogWriteAction() {
+  const inputSchema = convertZodtoJson(
+    z.object({
+      filePath: z.string().optional().describe('Defaults to catalog-info.yaml'),
+      entity: z
+        .string()
+        .optional()
+        .describe('You can provide the same values used in the Entity schema.'),
+    }),
+  );
+
   return createTemplateAction<{ filePath?: string; entity: Entity }>({
```

Comment:
Think the idea would be that we can remove or replace these types too, as they can be inferred from the zod schema or the response from the `inputSchema` like we have the `typeof schema.type` here: https://github.com/backstage/backstage/blob/da25293910f208d00296af7c7de497a8f8ff7278/plugins/scaffolder/src/components/fields/EntityTagsPicker/schema.ts#L16-L48